src/inmanta/execute/runtime.py

```diff
@@ -789,11 +789,14 @@ def for_namespace(self, namespace: Namespace) -> "Resolver":
         return NamespaceResolver(self, namespace)
 
     def get_dataflow_node(self, name: str) -> "dataflow.AssignableNodeReference":
-        graph: Optional[DataflowGraph] = self.dataflow_graph
-        if self.get_root_resolver() == self:
-            graph = self.namespace.get_scope().dataflow_graph
-        assert graph is not None
-        return graph.get_named_node(name)
+        try:
+            result_variable: Typeorvalue = self.lookup(name)
+            assert isinstance(result_variable, ResultVariable)
+            return result_variable.get_dataflow_node()
+        except NotFoundException:
+            root_graph: Optional[DataflowGraph] = self.get_root_resolver().dataflow_graph
+            assert root_graph is not None
+            return root_graph.get_own_variable(name)
```

Comment:
do I understand that all 'ownvariables' are in the root namespace, i.e. they collide?

Reply:
The first part, yes, because they aren't declared anywhere it's impossible to make a guess at scoping. I'm not sure what you mean with them colliding.

Reply:
if I have a for loop, with 500 iterations, each referring to an undefined 'x' it will be the same x

Reply:
Correct

Reply:
I would also comment here that this block is only executed if there is a reference to an undefined variable, and as we don't know in which scope it is defined (as it is undefined), we assume it is in the top scope to make our lives easier.